tests/services/apiService.test.js

```diff
@@ -40,3 +40,113 @@ describe("fetchQuestCategoryData function", () => {
     expect(result).toEqual(mockResponse);
   });
 });
+
+describe("fetchLeaderboardRankings function", () => {
+  beforeEach(() => {
+    fetch.mockClear();
+  });
+
+  it("should fetch and return data for valid parameters", async () => {
+    // Mock data creation for a successful response
+    const mockData = {
+      rankings: [
+        { address: '0x123abc', xp: 100, achievements: 5 },
+        { address: '0x456def', xp: 80, achievements: 3 },
+        { address: '0x789ghi', xp: 120, achievements: 7 },
+      ],
+      first_elt_position: 1,
+    };
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockData),
+    });
+
+    const params = { addr: '', page_size: 10, shift: 0, duration: 'week' };
+    const result = await fetchLeaderboardRankings(params);
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_ranking?addr=&page_size=10&shift=0&duration=week`
+    );
+    expect(result).toEqual(mockData);
+  });
+
+  it("should handle API returning no response", async () => {
+    // Mock fetch response with no data
+    fetch.mockResolvedValueOnce(undefined);
+
+    const params = { addr: 'invalidAddr', page_size: 10, shift: 0, duration: 'week' };
+    const result = await fetchLeaderboardRankings(params);
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_ranking?addr=invalidAddr&page_size=10&shift=0&duration=week`
+    );
+    expect(result).toBeUndefined();
+  });
+
+  it("should handle API returning response in unexpected format", async () => {
+    // Mock fetch response with unexpected data format
+    const mockResponse = "Unexpected response format";
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockResponse),
+    });
+
+    const params = { addr: 'sampleAddr', page_size: 10, shift: 0, duration: 'week' };
+    const result = await fetchLeaderboardRankings(params);
+
+    expect(fetch).toHaveBeenCalledWith(
+      `${API_URL}/leaderboard/get_ranking?addr=sampleAddr&page_size=10&shift=0&duration=week`
+    );
+    expect(result).toEqual(mockResponse);
+  });
+
+  it("should handle undefined cases in parameters", async () => {
+    const mockData = {
+      rankings: [
+        { address: '0x123abc', xp: 100, achievements: 5 },
+        { address: '0x456def', xp: 80, achievements: 3 },
+        { address: '0x789ghi', xp: 120, achievements: 7 },
+      ],
+      first_elt_position: 1,
+    };
+
+    fetch.mockResolvedValueOnce({
+      json: () => Promise.resolve(mockData),
+    });
+
+    const params1 = { addr: undefined, page_size: 10, shift: 0, duration: 'week' };
```

Comment:
this is only for one field i think. Also try undefined values for the other fields